README.md

```diff
@@ -1277,6 +1277,171 @@ $providerData->modifiedAt; //-> 2014-04-01T17:00:09.189Z
 $providerData->providerId; //-> facebook
 ```
 
+## Using Stormpath for API Authentication
+
+### Create an Account for your developers
+
+First, you will need user accounts in Stormpath to represent the people that are developing against 
+your API. Accounts can not only represent Developers, but also can be used to represent services, 
+daemons, processes, or any “entity” that needs to login to a Stormpath-secured API.
+
+```php
+$account = \Stormpath\Resource\Account::instantiate(
+  array('givenName' => 'Joe',
+        'surname' => 'Stormtrooper',
+        'email' => 'tk421@stormpath.com',
+        'password' => 'Changeme1'));
+
+$application->createAccount($account);
+```
+
+### Create and Manage API Keys for an Account
+
+After you create an account for a developer, you will need to generate an API Key (or multiple) 
+to be used when accessing your API. Each account will have an `apiKeys` property that contains a 
+collection of their API Keys. There will also be a list of API keys on a account’s profile in the 
+Stormpath Admin Console. You will be able to both create and manage keys in both.
+
+#### CREATING API KEYS FOR AN ACCOUNT 
+
+```php
+$apiKey = $account->createApiKey();
+
+$apiKeyId = $apikey->id;
+$apiKeySecret = $apikey->secret;
+```
+
+The `ApiKey` returned will have the following properties:
+* id	    The unique identifier for the API Key
+* secret	The secret for the API key.
+* status	A property that represent the status of the key. Keys with a disabled status will not be able to authenticate.
+* account	A link to the ApiKey’s account.
+* tenant	A link to the ApiKey’s tenant.
+
+#### MANAGE API KEYS FOR AN ACCOUNT
+
+##### Deleting an API Key
+
+```php
+$apiKeyId = 'FURThLWDE4MElDTFVUMDNDTzpQSHozZ';
+$apiKey = $application->getApiKey($apiKeyId);
+$apiKey->delete();
+```
+
+##### Disable an API key
+
+```php
+$apiKeyId = 'FURThLWDE4MElDTFVUMDNDTzpQSHozZ' 
+$apiKey = $application->getApiKey('FURThLWDE4MElDTFVUMDNDTzpQSHozZ');
+$apiKey->status = 'DISABLED';
+$apiKey->save()
+```
+
+### Using the Stormpath SDK to Authenticate and Generate Tokens for your API Keys
+
+The class `ApiRequestAuthenticator` can be used to perform both Basic Authentication 
+and OAuth Authentication.
+ 
+### Basic Authentication
+
+```
+GET /troopers/tk421/equipment 
+Accept: application/json
+Authorization: Basic MzRVU1BWVUFURThLWDE4MElDTFVUMDNDTzpQSHozZitnMzNiNFpHc1R3dEtOQ2h0NzhBejNpSjdwWTIwREo5N0R2L1g4
+Host: api.trooperapp.com
+```
+
+The `Authorization` header contains a base64 encoding of the API Key and Secret.
+
+#### Using `ApiRequestAuthenticator`
+
+```php
+$application = \Stormpath\Resource\Application::get("https://api.stormpath.com/v1/applications/24mp4us71ntza6lBwlu");
+
+$request = \Stormpath\Authc\Api\Request::createFromGlobals();
```

Comment:
How is the `createFromGlobals` method going to create the request? Also, does it make sense that the user has the option to provide the information for creating the request (like an options array as an argument)?